src/metabase/models/field_values.clj

```diff
@@ -138,28 +153,42 @@
 
 (t2/define-before-insert :model/FieldValues
   [{:keys [field_id] :as field-values}]
-  (u/prog1 (merge {:type :full}
-                  field-values)
+  (u/prog1 (merge {:type :full} field-values)
     (assert-valid-human-readable-values field-values)
-    (assert-valid-field-values-type field-values)
-    ;; if inserting a new full fieldvalues, make sure all the advanced field-values of this field is deleted
-    (when (= (:type <>) :full)
+    (assert-valid-type-hash-combo field-values)
+    ;; if inserting a new full fieldvalues, make sure all the advanced field-values of this field are deleted
+    (when (= :full (keyword (:type <>)))
```

Comment:
Was this a bug? should it have a test?

Reply:
Yip, added test

Reply:
Is this a firm convention? For non-test code the approach I take is to put the primitive expression first since it's much easier to parse. With the only layout you need to count brackets.

Reply:
It's probably not a firm convention but I just find it slightly surprising otherwise. I hardly ever see non-test code that does it this way. More than two brackets is rare to see anyway, so maybe your point (which is valid) doesn't really matter that much.

Reply:
Just to make sure I understand the convention in your head, does this capture it correctly?

"When comparing an expression that depends on some variables, to a constant expression, put the context expression last"

Reply:
yeah